objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,46 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)getAnyTypeName {
+  NSAssert([self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix],
+           @"Invalid any type url (%@).", self.typeURL);
+  return [self.typeURL substringFromIndex:[GPBTypeGoogleApisComPrefix length]];
+}
+
+- (instancetype)initWithMessage:(GPBMessage*)message {
+  self = [super init];
+  if (self) {
+    [self setMessage:message];
+  }
+  return self;
+}
+
+- (void)setMessage:(GPBMessage*)message {
+  self.typeURL = [GPBTypeGoogleApisComPrefix stringByAppendingString:message.descriptor.name];
+  self.value = message.data;
+}
+
+- (GPBMessage*)messageOfClass:(Class)messageClass {
+  if ([[self getAnyTypeName] isEqualToString:messageClass.descriptor.name]) {
+    GPBMessage* message = [messageClass message];
+    [message mergeFromData:self.value extensionRegistry:nil];
+    return message;
+  } else {
+    return nil;
+  }
+}
+
+- (BOOL)wrapsMessageOfClass:(Class)messageClass {
+  NSAssert([messageClass isSubclassofClass:[GPBMessage class]],
```

Comment:
If passing an invalid messageClass is considered programmer error, I prefer to only have the assert. (1. They can find such error in debug mode, and they should enable debug mode when debugging. 2. Returning NO implies this not an programmer error. Users who never enable debug mode will think it's a correct behavior to passing invalid messageClass).

The question is do we treat is as a programmer error or not?